src/test/java/api/requests/scenarios/HoldShelfExpirationDateTests.java

```diff
@@ -0,0 +1,188 @@
+package api.requests.scenarios;
+
+import static api.support.builders.RequestBuilder.OPEN_AWAITING_PICKUP;
+import static api.support.builders.RequestBuilder.OPEN_IN_TRANSIT;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyOrNullString;
+
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.circulation.support.ClockManager;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.ISODateTimeFormat;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import api.support.APITests;
+import api.support.builders.CheckInByBarcodeRequestBuilder;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+
+@RunWith(JUnitParamsRunner.class)
+public class HoldShelfExpirationDateTests extends APITests{
+  private static Clock clock;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    clock = Clock.fixed(Instant.now(), ZoneOffset.UTC);
+    ClockManager.getClockManager().setClock(clock);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    // reset the clock before each test (just in case)
+    ClockManager.getClockManager().setClock(clock);
+  }
+
+  @Test
+  @Parameters({
+    "cd5|MINUTES|42",
+    "cd6|HOURS|9",
+    "cd1|DAYS|30",
+    "cd2|MONTHS|6",
+    "cd4|WEEKS|2"
+  })
+  public void requestWithHoldShelfExpirationDateTest(String servicePoint, ChronoUnit interval, int amount)
```

Comment:
I think the postfix `Test` is redundant for tests, but again, that is likely stylistic and probably needs agreement on conventions.

Reply:
Holy guacamole! I didn't even notice other tests didn't have the `Test` postfix. I guess old habits die hard.

Reply:
In other tests (at least ones I've written, and haven't rushed) the name is intended to summarise the scenario being described (this is sometimes hard to do).

JUnit 5 brings some advantages in this area. And languages like Kotlin, Scala and Groovy (and tools like Spock) have major advantages for this style.